cmd/govim/main.go

```diff
@@ -349,7 +349,17 @@ func goModPath(wd string) (string, error) {
 		return "", fmt.Errorf("failed to execute [%v] in %v: %v\n%s", strings.Join(cmd.Args, " "), wd, err, out)
 	}
 
-	return strings.TrimSpace(string(out)), nil
+	gomod := strings.TrimSpace(string(out))
+	base := filepath.Base(gomod)
+	dir := filepath.Dir(gomod)
+
+	// Follow symlinks for the directory to match what vim returns for getcwd().
```

Comment:
Small nit, but the change here could sit within `newModWatcher` couldn't/shouldn't it? Because _that's_ the place where symlinks are not followed. i.e. we have to have evaluated symlinks prior to creating the watcher, else we would need to configure a watching that follows symlinks (which we might want to do in any case?)

Reply:
Your comment made me think about this another round since it is actually the inconsistence between vim's `getcwd()` and what `go env GOMOD` returns.

Turns out that it is `exec.Command` that won't use provided `cmd.Dir` if it points to the same dir as `$PWD`, even if one is the "real path" and the other is sym-linked.

Heschi pointed out in #tools that [x/tools handle this](https://github.com/golang/tools/blob/fe37c9e135b934191089b245ac29325091462508/internal/gocommand/invoke.go#L208) by adding `$PWD` so I went for that solution here as well. Then we'll get the same behaviour as gopls.